code/game/machinery/kitchen/smartfridge.dm

```diff
@@ -128,194 +126,253 @@
 
-	if(item_quants[O.name])
-		item_quants[O.name]++
+/obj/structure/machinery/smartfridge/proc/add_item(var/list/target, var/obj/item/O)
+	O.forceMove(src)
+	if(target[O.name])
+		target[O.name] += O
 	else
-		item_quants[O.name] = 1
+		target[O.name] = list(O)
 
 /obj/structure/machinery/smartfridge/proc/add_network_item(var/obj/item/O)
 	if(is_in_network())
-		chemical_data.shared_item_storage.Add(O)
-
-		if(chemical_data.shared_item_quantity[O.name])
-			chemical_data.shared_item_quantity[O.name]++
-		else
-			chemical_data.shared_item_quantity[O.name] = 1
+		add_item(chemical_data.shared_item_storage, O)
 		return TRUE
 	return FALSE
 
 //*******************
 //*   SmartFridge Menu
 //********************/
 
-/obj/structure/machinery/smartfridge/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
-	user.set_interaction(src)
-
-	var/data[0]
-	data["contents"] = null
-	data["wires"] = null
-	data["panel_open"] = panel_open
-	data["electrified"] = seconds_electrified > 0
-	data["shoot_inventory"] = shoot_inventory
-	data["locked"] = locked
-	data["secure"] = is_secure_fridge
-	data["networked"] = is_in_network()
-	data["transfer_mode"] = transfer_mode
-
-	var/list/items[0]
-	for (var/i=1 to length(item_quants))
-		var/K = item_quants[i]
-		var/count = item_quants[K]
-		if (count > 0)
-			items.Add(list(list("display_name" = html_encode(capitalize(K)), "vend" = i, "quantity" = count)))
+/obj/structure/machinery/smartfridge/tgui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if (!ui)
+		ui = new(user, src, "SmartFridge", name)
+		ui.open()
 
-	if (length(items) > 0)
-		data["contents"] = items
+/obj/structure/machinery/smartfridge/ui_state(mob/user)
+	return GLOB.not_incapacitated_and_adjacent_strict_state
 
-	if(is_in_network())
-		var/list/networked_items = list()
-		for (var/i=1 to length(chemical_data.shared_item_quantity))
-			var/K = chemical_data.shared_item_quantity[i]
-			var/count = chemical_data.shared_item_quantity[K]
-			if (count > 0)
-				networked_items.Add(list(list("display_name" = html_encode(capitalize(K)), "vend" = i, "quantity" = count)))
+/obj/structure/machinery/smartfridge/ui_status(mob/user, datum/ui_state/state)
+	. = ..()
+	if(inoperable())
+		return UI_CLOSE
+
+/obj/structure/machinery/smartfridge/ui_static_data(mob/user)
+	. = ..(user)
+	.["networked"] = is_in_network()
 
-		if (length(networked_items) > 0)
-			data["networked_contents"] = networked_items
+/obj/structure/machinery/smartfridge/ui_data(mob/user)
+	. = list()
+	.["secure"] = is_secure_fridge
+	.["transfer_mode"] = transfer_mode
+	.["locked"] = locked
 
 	var/list/wire_descriptions = get_wire_descriptions()
 	var/list/panel_wires = list()
 	for(var/wire = 1 to wire_descriptions.len)
 		panel_wires += list(list("desc" = wire_descriptions[wire], "cut" = isWireCut(wire)))
 
-	if (panel_wires)
-		data["wires"] = panel_wires
+	.["electrical"] = list(
+		"electrified" = seconds_electrified > 0,
```

Comment:
I am going to invoke the "you touch it you fix it" clause to say PLEASE refactor the seconds_electrified thing to use actual timers/cooldowns rather than /process

Reply:
Done using cooldowns